### PR TITLE
[Backport][2.11] Backport patches in test/fuzz

### PR DESCRIPTION
**Pull request by ligurio**
- grammar SQL fuzzer
- grammar Lua fuzzer
- fixes-shmixes for fuzzing tests
- fix for https://github.com/tarantool/tarantool/issues/8502

### CODE REVIEW

File: changelogs/unreleased/luajit-grammar-fuzzer.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## feature/test/fuzz
+
+* LuaJIT now can be fuzzed using grammar-based fuzzer (gh-4823).
```
Comment (andreyaksenov) [inline changelogs/unreleased/luajit-grammar-fuzzer.md:3]:
```suggestion
* LuaJIT can now be fuzzed using a grammar-based fuzzer (gh-4823).
```

**Review by andreyaksenov — APPROVED** [review]

File: changelogs/unreleased/luajit-grammar-fuzzer.md (hunk 1 above)
Comment (ligurio) [reply changelogs/unreleased/luajit-grammar-fuzzer.md:3]: it's a backport

**Comment by igormunkin** [comment]
@ligurio, thanks for the patchset! I would like to notice, that the cherry-picked patches must contain the original hash in the commit message (i.e. use `cherry-pick -x` for backporting).

File: changelogs/unreleased/luajit-grammar-fuzzer.md (hunk 1 above)
Comment (igormunkin) [reply changelogs/unreleased/luajit-grammar-fuzzer.md:3]: Well... I also agree, that this is a backport of the patch and it should be taken intact. OTOH, this patch has been committed prior to doc team is summoned to each PR to reduce changelog fixups when release comes. What I definitely don't want, that's typos in the resulting changelog. @andreyaksenov, @veod32, can we omit your suggestions for now, or you're not going to fix any grammar while tagging the release?

Comment (igormunkin) [reply changelogs/unreleased/luajit-grammar-fuzzer.md:3]: OK, we'll proceed with no changes then.

**Review by igormunkin — APPROVED** [review]
@ligurio, thanks for the fixes! References for the three top-most commits are invalid:
* `test/fuzz: add breaks to switch-case`
* `test/fuzz: fix luaJIT fuzzer timeout`
* `test/fuzz: refactor LuaJIT fuzzer`

You used commit hashes from the PR, but the patches were cherry-picked, so the hash differs. LGTM otherwise.

**Pull request merged**